### PR TITLE
Feature/overflow tests

### PR DESCRIPTION
**Pull request by goncer**

### CODE REVIEW

File: test/parallel/xyk-pallet.API.overflow.test.ts [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+
+	
+		// check users accounts.
+		await testUser1.refreshAmounts(AssetWallet.BEFORE);
+
+	});
+	test('Create pool of [MAX,1]: OverFlow [a+b] - liquidityAsset calculation', async () => {
+		await waitNewBlock();
+
+		await createPool(testUser1.keyRingPair ,secondCurrency, new BN(1), firstCurrency, MAX_BALANCE)
```
Comment (goncer) [inline test/parallel/xyk-pallet.API.overflow.test.ts:58]: Hi @iStrike7 ,
On this file, I use the MAX_BALANCE without any issue

File: test/exploratory/RPC.bigNumber.test.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+		
+		// add users to pair.
+		keyring.addPair(testUser1.keyRingPair);
+		keyring.addPair(sudo.keyRingPair);
+		api = getApi();
+	});
+	const MAX_BALANCE = new BN('340282366920938463463374607431768211455'); //max balance
+	test('Request big numbers', async () => {
+
+		let result = await ( api.rpc as any).xyk.calculate_buy_price(MAX_BALANCE, MAX_BALANCE, MAX_BALANCE);
```
Comment (goncer) [inline test/exploratory/RPC.bigNumber.test.ts:55]: hi @iStrike7 
When using MAX_BALANCE it in the RPC it fails badly,
```
    2021-07-09 08:17:47        RPC-CORE: calculate_buy_price(input_reserve: Balance, output_reserve: Balance, sell_amount: Balance): RpcResult<Balance>:: -32602: Invalid params: invalid type: string "0xfffffffffffffffffffffffffffffffe", expected u128.

      at apply (node_modules/@polkadot/util/logger.js:63:22)
      at Object.error (node_modules/@polkadot/util/logger.js:98:27)
      at logErrorMessage (node_modules/@polkadot/rpc-core/index.js:63:5)
      at node_modules/@polkadot/rpc-core/index.js:270:11
```

Comment (goncer) [reply test/exploratory/RPC.bigNumber.test.ts:52]: @iStrike7 , any news from the RPCs calls? were you able to investigate it?
Thanks

File: test/parallel/xyk-rpc.errors.test.ts [hunk 3]
```diff
@@ -117,4 +118,25 @@ test.each([
 
 
 
+test.each([
+	[new BN('100000000')],
+	[new BN('100000000000000')],
+	[new BN('10000000000000000000000')] //<- Fails with:
```
Comment (goncer) [inline test/parallel/xyk-rpc.errors.test.ts:124]: @mateuszaaa ,
following your comment in trello, I thought that there is hope, but still failing :(
I run out of ideas and I think @iStrike7  too.
any clue ?

File: test/parallel/xyk-pallet.API.overflow.test.ts (hunk 1 above)
Comment (goncer) [reply test/parallel/xyk-pallet.API.overflow.test.ts:58]: issue was solved by Mat PR.

File: test/parallel/xyk-rpc.errors.test.ts (hunk 3 above)
Comment (goncer) [reply test/parallel/xyk-rpc.errors.test.ts:124]: solved by Mat.

File: test/exploratory/RPC.bigNumber.test.ts (hunk 2 above)
Comment (goncer) [reply test/exploratory/RPC.bigNumber.test.ts:53]: Mat fixed this :)

**Comment by goncer** [comment]
Relevant changes, 
Others are refactors regarding using BN instead of numbers :)

Big number test :
https://github.com/mangata-finance/mangata-e2e/pull/24/files?file-filters%5B%5D=.lock&file-filters%5B%5D=.ts#diff-342b7c12318de8cb64fad16a4aa544e234d12fb985ee280adaae3692865e001d

Overflow tests:
https://github.com/mangata-finance/mangata-e2e/pull/24/files?file-filters%5B%5D=#diff-c8c255a63bb1d0c8d9a54d6160d3d25cc65ab0a8cf68c86c59769f728fd494bf

Fix for debug tests:
https://github.com/mangata-finance/mangata-e2e/pull/24/files?file-filters%5B%5D=.lock&file-filters%5B%5D=.ts#diff-cd973153292765c95ab71e093846ebd4c9554da21b318dc83eea7dd494bb44e0

fix for pools that does not exist:

https://github.com/mangata-finance/mangata-e2e/pull/24/files?file-filters%5B%5D=.lock&file-filters%5B%5D=.ts#diff-c7b01ba73caeee9e0c424b9cbc65030d22db0d866a0a4254bb1f60db624c06b5

Enum of error codes:

https://github.com/mangata-finance/mangata-e2e/pull/24/files?file-filters%5B%5D=.lock&file-filters%5B%5D=.ts#diff-34b033806cd95cba13daa6c56736722fb1f3b9de424bb99ba34ed00336b00131

**Review by mateuszaaa — APPROVED** [review]
Great job

**Pull request merged**